geode-core/src/distributedTest/java/org/apache/geode/distributed/internal/P2PMessagingSSLTLSKeyUpdateDUnitTest.java

```diff
@@ -0,0 +1,368 @@
+      };
+
+      for (int i = 0; i < SENDER_COUNT; ++i) {
+        executor.submit(doSending);
+      }
+
+      stopLatch.await();
+
+      assertThat(failedRecipientCount.sum()).as("message delivery failed N times").isZero();
+    });
+
+    final long bytesSent = getByteCount(sender);
+
+    await().timeout(Duration.ofSeconds(10)).untilAsserted(
+        () -> assertThat(getByteCount(receiver))
+            .as("bytes received != bytes sent")
+            .isEqualTo(bytesSent));
+  }
+
+  private long getByteCount(final MemberVM member) {
+    return member.invoke(() -> bytesTransferredAdder.sum());
+  }
+
+  private static ClusterDistributionManager getCDM() {
+    return (ClusterDistributionManager) ((InternalCache) CacheFactory.getAnyInstance())
+        .getDistributionManager();
+  }
+
+  private static class TestMessage extends DistributionMessage {
+
+    /*
+     * When this comment was written, messageId wasn't used for anything.
+     * The field was added during a misguided attempt to add SHA-256
+     * digest verification on sender and receiver. Then I figured out
+     * that there's no way to parallelize that (for the sender) so
+     * I settled for merely validating the number of bytes transferred.
+     * Left the field here in case it comes in handy later.
+     */
+    private volatile int messageId;
+
+    TestMessage(final InternalDistributedMember receiver,
+        final int messageId) {
+      setRecipient(receiver);
+      this.messageId = messageId;
+    }
+
+    // necessary for deserialization
+    public TestMessage() {
+      messageId = 0;
+    }
+
+    @Override
+    public int getProcessorType() {
+      return OperationExecutors.STANDARD_EXECUTOR;
+    }
+
+    @Override
+    protected void process(final ClusterDistributionManager dm) {}
+
+    @Override
+    public void toData(final DataOutput out, final SerializationContext context)
+        throws IOException {
+      super.toData(out, context);
+
+      out.writeInt(messageId);
+
+      final ThreadLocalRandom random = ThreadLocalRandom.current();
+      final int length = MESSAGE_SIZE;
+
+      out.writeInt(length);
+
+      final byte[] payload = new byte[length];
+      random.nextBytes(payload);
+
+      out.write(payload);
+
+      /*
+       * the LongAdder should ensure that we don't introduce any (much)
+       * synchronization with other concurrent tasks here
+       */
+      bytesTransferredAdder.add(length);
+    }
+
+    @Override
+    public void fromData(final DataInput in, final DeserializationContext context)
+        throws IOException, ClassNotFoundException {
+      super.fromData(in, context);
+
+      messageId = in.readInt();
+
+      final int length = in.readInt();
+
+      final byte[] payload = new byte[length];
+
+      in.readFully(payload);
+
+      bytesTransferredAdder.add(length);
+    }
+
+    @Override
+    public int getDSFID() {
+      return NO_FIXED_ID; // for testing only!
+    }
+  }
+
+  @NotNull
```

Comment:
While technically a legal place to put the annotation, I think it is "more correct" to have it next to the type rather than the method modifiers. This say way you know the annotation applies to the return value and isn't a general method modifier annotation.
```java
private static @NotNull Properties geodeConfigurationProperties() ...
```

Reply:
fixed it